core/state/state_object.go

```diff
@@ -359,15 +358,12 @@ func (s *stateObject) updateTrie() (Trie, error) {
 		// Cache the items for preloading
 		usedStorage = append(usedStorage, common.CopyBytes(key[:])) // Copy needed for closure
 	}
-	if len(keysToDelete) > 0 {
-		sort.Slice(keysToDelete, func(i, j int) bool { return bytes.Compare(keysToDelete[i][:], keysToDelete[j][:]) < 0 })
-		for _, key := range keysToDelete {
-			if err := tr.DeleteStorage(s.address, key[:]); err != nil {
-				s.db.setError(err)
-				return nil, err
-			}
-			s.db.StorageDeleted += 1
+	for _, key := range deletions {
```

Comment:
This changes the order in which keys are deleted compared to previous version, which could is very important because the order affects record/replay (as it is, some blocks recording won't fit the replay).
The behavior only matters if s.deterministic is true, but in this case it should be consistent with the previous version.
"deletions" here seems to be identical to "keysToDelete" in the previous deterministic case.
If I'm right, all we need to do is check s.deterministic, and if it's true, before deleting keys we need to
```
sort.Slice(keysToDelete, func(i, j int) bool { return bytes.Compare(keysToDelete[i][:], keysToDelete[j][:]) < 0 })
```
and that will be consistent with previous behaviour.

Reply:
Done.